### PR TITLE
chore: add standard test logger ignoring db canceled

### PR DESCRIPTION
**Pull request by spikecurtis**
Refactors our use of `slogtest` to instantiate a "standard logger" across most of our tests.  This standard logger incorporates https://github.com/coder/slog/pull/217 to also ignore database query canceled errors by default, which are a source of low-severity flakes.

Any test that has set non-default `slogtest.Options` is left alone. In particular, `coderdtest` defaults to ignoring all errors. We might consider revisiting that decision now that we have better tools to target the really common flaky Error logs on shutdown.

### CODE REVIEW

**Comment by spikecurtis** [comment]
* **#15556** <a href="https://app.graphite.dev/github/pr/coder/coder/15556?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/coder/coder/15556?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by johnstcn — COMMENTED** [review]
I'm a fan of this overall approach of extracting commonly-used test apparatus into `testutil`. I think your refactor missed some imports in `tailnet/test/integration` but I'm happy to approve once those are fixed and tests are passing.

File: provisioner/terraform/install.go [hunk 1]
```diff
@@ -49,9 +49,13 @@ func Install(ctx context.Context, log slog.Logger, dir string, wantVersion *vers
...
+	hasVersionStr := "nil"
 	hasVersion, err := versionFromBinaryPath(ctx, binPath)
-	if err == nil && hasVersion.Equal(wantVersion) {
-		return binPath, err
+	if err == nil {
+		hasVersionStr = hasVersion.String()
+		if hasVersion.Equal(wantVersion) {
+			return binPath, err
+		}
 	}
```
Comment (johnstcn) [inline provisioner/terraform/install.go:59]: Should this be in a separate PR?

**Review by johnstcn — APPROVED** [review]
I only have that one comment regarding `provisioner/terraform/install.go` but that's not blocking.

File: provisioner/terraform/install.go (hunk 1 above)
Comment (spikecurtis) [reply provisioner/terraform/install.go:59]: It's true that it's a bug fix. Tests weren't exercising the `Debug` log prior to this, so we never caught that this could segfault.

I can strip it out into it's own PR if you like.

Comment (johnstcn) [reply provisioner/terraform/install.go:59]: I'm fine leaving it here!

**Comment by spikecurtis** [comment]
### Merge activity

* **Nov 18, 5:09 AM EST**: A user merged this pull request with [Graphite](https://app.graphite.dev/github/pr/coder/coder/15556).
